src/plugins/saved_objects/public/saved_object/saved_object.test.ts

```diff
@@ -94,26 +96,67 @@ describe('Saved Object', () => {
    * @returns {Promise<SavedObject>} A promise that resolves with an instance of
    * SavedObject
    */
-  function createInitializedSavedObject(config: SavedObjectConfig = {}) {
+  function createInitializedSavedObject(config: SavedObjectConfig = { type: 'dashboard' }) {
     const savedObject = new SavedObjectClass(config);
     savedObject.title = 'my saved object';
 
     return savedObject.init!();
   }
 
-  beforeEach(() => {
-    SavedObjectClass = createSavedObjectClass(({
-      savedObjectsClient: savedObjectsClientStub,
-      indexPatterns: dataStartMock.indexPatterns,
-      search: {
-        ...dataStartMock.search,
-        searchSource: {
-          ...dataStartMock.search.searchSource,
-          create: createSearchSourceMock,
-          createEmpty: createSearchSourceMock,
+  const initSavedObjectClass = () => {
+    SavedObjectClass = createSavedObjectClass(
+      ({
+        savedObjectsClient: savedObjectsClientStub,
+        indexPatterns: dataStartMock.indexPatterns,
+        search: {
+          ...dataStartMock.search,
+          searchSource: {
+            ...dataStartMock.search.searchSource,
+            create: createSearchSourceMock,
+            createEmpty: createSearchSourceMock,
+          },
         },
-      },
-    } as unknown) as SavedObjectKibanaServices);
+      } as unknown) as SavedObjectKibanaServices,
+      decoratorRegistry
+    );
+  };
+
+  beforeEach(() => {
+    decoratorRegistry = savedObjectsDecoratorRegistryMock.create();
+    initSavedObjectClass();
+  });
+
+  describe('decorators', () => {
+    it('calls the decorators during construct', () => {
+      const decorA = {
+        getId: () => 'A',
+        decorateConfig: jest.fn(),
+        decorateObject: jest.fn(),
+      };
+      const decorB = {
+        getId: () => 'B',
+        decorateConfig: jest.fn(),
+        decorateObject: jest.fn(),
+      };
+
+      decoratorRegistry.getOrderedDecorators.mockReturnValue([decorA, decorB]);
+
+      initSavedObjectClass();
+      createInitializedSavedObject();
+
+      expect(decorA.decorateConfig).toHaveBeenCalledTimes(1);
```

Comment:
could you add a test that changes (mutations) done in the first decorator are available in the second one?